### PR TITLE
[docs] Add Release Notes document

### PR DESCRIPTION
**Pull request by roel4ez**
This PR adds release notes for v2.0.0
#1246

### CODE REVIEW

File: docs/release-notes.md [hunk 1]
```diff
@@ -0,0 +1,44 @@
+---
+title: Release Notes
+---
+
+## v2.0.0
+
+### New Features
+
+- Support for multiple Azure IoTEdge Gateways with cache and deduplication
```
Comment (Mandur) [inline docs/release-notes.md:9]:
```suggestion
- Support for multiple Azure IoT Edge Gateways with cache and deduplication
```

File: docs/release-notes.md [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+- Support for multiple concentrators with cache and deduplication
+- Support for running LNS on a separate Edge or Cloud device
+- [EFLOW](https://docs.microsoft.com/en-us/azure/iot-edge/iot-edge-for-linux-on-windows?view=iotedge-2018-06) Compatibility
```
Comment (Mandur) [inline docs/release-notes.md:12]:
```suggestion
- [Support for LNS on Windows](https://docs.microsoft.com/en-us/azure/iot-edge/iot-edge-for-linux-on-windows?view=iotedge-2018-06) Compatibility
```

File: docs/release-notes.md [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+- Support for [LoRa Basics™ Station](https://github.com/lorabasics/basicstation)
+- Regional support for CN470 and AS923
```
Comment (Mandur) [inline docs/release-notes.md:14]: Should we be explicit about RP1 and RP2 for china?

File: docs/release-notes.md [hunk 4]
```diff
@@ -0,0 +1,44 @@
+---
+title: Release Notes
+---
+
+## v2.0.0
+
+### New Features
+
```
Comment (danigian) [inline docs/release-notes.md:8]: - Support for [LoRa Basics™ Station](https://github.com/lorabasics/basicstation/tree/v2.0.5) LNS Protocol (v2.0.5)
- Support for [LoRa Basics™ Station](https://github.com/lorabasics/basicstation/tree/v2.0.5) CUPS Protocol (v2.0.5) for credential management and firmware upgrade
- LoRaWAN Network Server and Basics™ Station can be decoupled, allowing the possibility to run multiple concentrators on a single LoRaWAN Network Server instance
- Support for secure communication between LoRaWAN Network Server and Basics™ Station
- Support for running LoRaWAN Network Server on Azure IoT Edge for Linux on Windows
- Regional support for AS923 and CN470 (Regional Parameters v1.0.3rA and RP002-1.0.3 revisions)
- Observability: Integration with Azure Monitor
- Observability: exposing a Prometheus endpoint for metrics scraping
- Observability: @bastbu please add more here 

I would also add a "Other" or "Code quality" section with:
- Introduction of LoRaWAN Primitives and refactoring of existing code to make wide use of those
- Upgrade to .NET 6 (LTS version)
- Upgrade container base images to Debian 11
- Improvements in E2E CI, continuously testing multiple scenarios and authentication modes
- @p-schuler is there something we want to write here or in the breaking changes regarding the device caching?

File: docs/release-notes.md [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+- Support for running LNS on a separate Edge or Cloud device
+- Support for running the LNS on [Windows](https://docs.microsoft.com/en-us/azure/iot-edge/iot-edge-for-linux-on-windows?view=iotedge-2018-06) 
+- Support for [LoRa Basics™ Station](https://github.com/lorabasics/basicstation)
+- Regional support for CN470 and AS923
+- Support for CUPS credential management
+- Support for CUPS concentrator firmware update
+- First class support for Class C devices
+- Observability: integration with Azure Monitor and metrics that are exposed on a Prometheus endpoint
+
+### Breaking Changes
```
Comment (danigian) [inline docs/release-notes.md:20]: In this section I would at least add:

- Infrastructure changes: New storage containers for CUPS credentials management and firmware upgrade (more in related [ADRs](link to ADRs)
- Infrastructure changes: @bastbu were there some changes related to deployment of Azure Monitor resources?

File: docs/release-notes.md (hunk 4 above)
Comment (p-schuler) [reply docs/release-notes.md:8]: > LoRaWAN Network Server and Basics™ Station can be decoupled, allowing the possibility to run multiple concentrators on a single LoRaWAN Network Server instance

Correction: allowing the possibility to **connect** multiple concentrators to a single LNS.

File: docs/release-notes.md (hunk 5 above)
Comment (p-schuler) [reply docs/release-notes.md:20]: We need log analytics for the integration with the metrics collector.

File: docs/release-notes.md (hunk 4 above)
Comment (p-schuler) [reply docs/release-notes.md:8]: There are not breaking changes to the device caching. It's been changed and improved, but not breaking compatibility.

Comment (p-schuler) [reply docs/release-notes.md:8]: Observability: default monitoring for edgeAgent and edgeHub modules.

File: docs/release-notes.md [hunk 6]
```diff
@@ -0,0 +1,44 @@
+---
+title: Release Notes
+---
+
+## v2.0.0
+
+### New Features
+
+- Support for multiple Azure IoT Edge Gateways with cache and deduplication
```
Comment (p-schuler) [inline docs/release-notes.md:9]: we always supported multi gateway.

File: docs/release-notes.md (hunk 4 above)
Comment (p-schuler) [reply docs/release-notes.md:8]: Observability: distributed tracing

File: .markdown-link-check.jsonc [hunk 7]
```diff
@@ -23,5 +23,12 @@
         "replacement": "{{BASEURL}}/"
       }
     ],
-    "aliveStatusCodes": [200, 206]
+    "aliveStatusCodes": [200, 206],
```
Comment (danigian) [inline .markdown-link-check.jsonc:26]: @roel4ez can we make these changes a separate pr?

thanks

File: docs/release-notes.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+### Bugfixes
+
+- [#249](https://github.com/Azure/iotedge-lorawan-starterkit/issues/249)
+At least one resource deployment operation failed
+- [#310](https://github.com/Azure/iotedge-lorawan-starterkit/issues/310)
+Incorrect data type for the tmms property
+
+## v2.0.0-beta
+
+- certificate validation
```
Comment (danigian) [inline docs/release-notes.md:58]: This is a breaking change:

Compared to v2.0.0-alpha, in v2.0.0-beta we start validating not only the certificate thumbprint (as per device twin) but also that the chain of trust is correct. Therefore, any user is required to follow the steps [here](https://azure.github.io/iotedge-lorawan-starterkit/dev/user-guide/station-authentication-modes/?h=chain#changing-client-certificate-mode-in-lorawan-network-server-module-and-trusting-certificate-chain)

File: docs/release-notes.md [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+- Support for running LoRaWAN Network Server on [Azure IoT Edge for Linux on
+Windows](https://docs.microsoft.com/en-us/azure/iot-edge/iot-edge-for-linux-on-windows?view=iotedge-2018-06)
+- Regional support for AS923 and CN470 (Regional Parameters v1.0.3rA and
+RP002-1.0.3 revisions)
+- Support for running LNS on a separate Edge or Cloud device
+- First class support for Class C devices
+- Observability:
+  - Integration with Azure Monitor
+  - Exposing a Prometheus endpoint for metrics scraping
+  - default monitoring for edgeAgent and edgeHub modules
```
Comment (danigian) [inline docs/release-notes.md:27]:
```suggestion
  - Collection of metrics for edgeAgent and edgeHub modules with [Metrics collector module](https://docs.microsoft.com/en-us/azure/iot-edge/how-to-collect-and-transport-metrics?view=iotedge-2020-11&tabs=iothub)
```

File: docs/release-notes.md [hunk 10]
```diff
@@ -0,0 +1,72 @@
...
+  - distributed tracing
```
Comment (danigian) [inline docs/release-notes.md:28]: We should be more detailed on what this means

File: docs/release-notes.md [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+possibility to connect multiple concentrators on a single LoRaWAN Network Server
+instance
+- Support for secure communication between LoRaWAN Network Server and Basics™
+Station
+- Support for running LoRaWAN Network Server on [Azure IoT Edge for Linux on
+Windows](https://docs.microsoft.com/en-us/azure/iot-edge/iot-edge-for-linux-on-windows?view=iotedge-2018-06)
+- Regional support for AS923 and CN470 (Regional Parameters v1.0.3rA and
+RP002-1.0.3 revisions)
+- Support for running LNS on a separate Edge or Cloud device
+- First class support for Class C devices
```
Comment (danigian) [inline docs/release-notes.md:23]:
```suggestion
```
I would remove this sentence. We already had Class C support before

File: .markdown-link-check.jsonc (hunk 7 above)
Comment (roel4ez) [reply .markdown-link-check.jsonc]: done, see #1427 - which would need to be merged before this.

File: docs/release-notes.md (hunk 10 above)
Comment (bastbu) [reply docs/release-notes.md:27]: I think we should remove it, we don't really have support for distributed tracing.

File: docs/release-notes.md [hunk 12]
```diff
@@ -0,0 +1,75 @@
...
+possibility to connect multiple concentrators on a single LoRaWAN Network Server
+instance
+- Support for secure communication between LoRaWAN Network Server and Basics™
+Station
+- Support for running LoRaWAN Network Server on [Azure IoT Edge for Linux on
+Windows](https://docs.microsoft.com/en-us/azure/iot-edge/iot-edge-for-linux-on-windows?view=iotedge-2018-06)
+- Regional support for AS923 and CN470 (Regional Parameters v1.0.3rA and
+RP002-1.0.3 revisions)
+- Support for running LNS on a separate Edge or Cloud device
+- Observability:
```
Comment (bastbu) [inline docs/release-notes.md:23]:
```suggestion
- [Observability](./user-guide/observability.md):
```

File: docs/release-notes.md [hunk 13]
```diff
@@ -0,0 +1,75 @@
...
+- Upgrade to .NET 6 (LTS version)
+- Upgrade container base images to Debian 11
+- Improvements in E2E CI, continuously testing multiple scenarios and
+authentication modes
+
+### Breaking Changes
+
+- Deprecated support for Packetforwarder ([upgrade instructions](user-guide/pkt-fwd-to-station.md))
+- Upgrade to RaspberryPi OS based on Debian 11.0 Bullseye ([upgrade instructions](user-guide/upgrade.md#upgrading-to-raspberry-pi-os-bullseye))
+- Infrastructure changes:
```
Comment (danigian) [inline docs/release-notes.md:42]: Let's add that the Azure Function extension version needs to be updated to ~4 to support v2.0.0

Comment (p-schuler) [reply docs/release-notes.md:42]: maybe link the upgrade instructions in general rather than repeating that information: https://azure.github.io/iotedge-lorawan-starterkit/dev/user-guide/upgrade/

**Review by danigian — APPROVED** [review]
